lib/Generator.js

```diff
@@ -20,40 +20,40 @@
       var chmod = Q.denodeify(fs.chmod);
       var mkdir = Q.denodeify(mkdirp);
 
-      //function that handles .gitignore
-      function gitignore(path){
-        return function(){
-          return fs.appendFile('.gitignore', "\n" + path, function(err){
-            if(err) return Promise.reject(err);
-            return Promise.resolve();
-          });
-        };
-      }
 
       // Kick off a promise chain
       mkdir('ansible/inventory', '0755')
-        .then(getReader('ansible.cfg.mustache')).then(render).then(getWriter('ansible/ansible.cfg'))
-        .then(getReader('aws_keys.mustache')).then(render).then(getWriter('ansible/inventory/aws_keys'))
-        .then(gitignore('ansible/inventory/aws_keys'))
-        .then(getReader('destroy-old-nodes.yaml.mustache')).then(render)
-          .then(getWriter('ansible/destroy-old-nodes.yaml'))
-        .then(getReader('ec2.ini.mustache')).then(render).then(getWriter('ansible/inventory/ec2.ini'))
-        .then(getReader('ec2.py')).then(render).then(getWriter('ansible/inventory/ec2.py'))
-        .then(getReader('immutable.yaml.mustache')).then(render).then(getWriter('ansible/immutable.yaml'))
-        .then(getReader('provision.sh.mustache')).then(render).then(getWriter('ansible/provision.sh'))
-          .then(chmod('ansible/provision.sh', '0744'))
-        .then(getReader('tag-old-nodes.yaml.mustache')).then(render).then(getWriter('ansible/tag-old-nodes.yaml'))
+        .then(process('ansible.cfg.mustache', 'ansible/ansible.cfg'))
+        .then(process('aws_keys.mustache', 'ansible/inventory/aws_keys'))
+        .then(process('destroy-old-nodes.yaml.mustache', 'ansible/destroy-old-nodes.yaml'))
+        .then(process('ec2.ini.mustache', 'ansible/inventory/ec2.ini'))
+        .then(process('ec2.py', 'ansible/inventory/ec2.py'))
+        .then(process('immutable.yaml.mustache', 'ansible/immutable.yaml'))
+        .then(process('provision.sh.mustache', 'ansible/provision.sh'))
+        .then(process('tag-old-nodes.yaml.mustache', 'ansible/tag-old-nodes.yaml'))
         .then(mkdir('ansible/roles/' + answers.appName + '/tasks'))
-        .then(getReader('roles/appName/tasks/main.yaml.mustache')).then(render).then(getWriter('ansible/roles/' +
-          answers.appName + '/tasks/main.yaml'))
-        .then(getReader('myApp.pem.mustache')).then(render).then(getWriter('ansible/' + answers.appPem))
-        .then(gitignore('ansible/' + answers.appPem))
```

Comment:
merge conflicts are pretty difficult to handle, I screw them up on a regular basis :-)

What I tend to do is, when I rebase, to check against my target branch and the remote version of my current branch:

```bash
# What's changed in my local branch vs the unrebased remote copy?
git diff origin/my-branch
# What's changed in my local branch vs my target?
git diff origin/master
```

Clever sausages like Warren and Nate probably have all kinds of fancy strategies involving hunks and partial application, but what I tend to do is eyeball it and paste in any lines I've accidentally removed from my IDE. I use PHPStorm and it's got some pretty sweet features for showing diffs and git history (which is why I didn't want the whitespace commit as it would break my workflow slightly and I am very, very lazy)

This is the beauty of squashing to a single commit - everyone thinks my code is awesome whereas, really, I screw it up like 5 or 6 times before I finish creating my pull request! ;-)